### ISSUE TITLE
Error: Unable to get elm-package.json

### ISSUE THREAD

**Issue by ormsbee** [evancz/start-app#36]
I tried to install this package to play around with it, and got the following error:

```
$ elm-package install evancz/start-app
To install evancz/start-app I would like to add the following
dependency to elm-package.json:

    "evancz/start-app": "2.0.1 <= v < 3.0.0"

May I add that to elm-package.json for you? (y/n) y


Error: Unable to get elm-package.json for evancz/start-app 2.0.1
Failed reading: satisfy
```

I am using: `Elm Package Manager 0.5.1 (Elm Platform 0.15.1)`

I'm a complete newb in all things elm, so apologies if I'm missing something very basic.

Take care.

**Comment by ormsbee** [comment]
FWIW, other packages (e.g. `elm-html`, `elm-effects`) install fine.

**Comment by jvoigtlaender** [comment]
If this is still a problem, can you post your existing `elm-package.json` here?

**Comment by ormsbee** [comment]
I started over with an empty dir if it helps:

```
$ mkdir helloworld1
$ cd helloworld1/
$ elm-package install evancz/start-app
To install evancz/start-app I would like to add the following
dependency to elm-package.json:

    "evancz/start-app": "2.0.1 <= v < 3.0.0"

May I add that to elm-package.json for you? (y/n) y


Error: Unable to get elm-package.json for evancz/start-app 2.0.1
Failed reading: satisfy

$ cat elm-package.json
{
    "version": "1.0.0",
    "summary": "helpful summary of your project, less than 80 characters",
    "repository": "https://github.com/USER/PROJECT.git",
    "license": "BSD3",
    "source-directories": [
        "."
    ],
    "exposed-modules": [],
    "dependencies": {
        "elm-lang/core": "2.1.0 <= v < 3.0.0",
        "evancz/start-app": "2.0.1 <= v < 3.0.0"
    },
    "elm-version": "0.15.1 <= v < 0.16.0"
}
```

By contrast, this is me making another empty dir and successfully installing `elm-html`

```
$ mkdir helloworld2
$ cd helloworld2
$ elm-package install evancz/elm-html
To install evancz/elm-html I would like to add the following
dependency to elm-package.json:

    "evancz/elm-html": "4.0.1 <= v < 5.0.0"

May I add that to elm-package.json for you? (y/n) y

Some new packages are needed. Here is the upgrade plan.

  Install:
    elm-lang/core 2.1.0
    evancz/elm-html 4.0.1
    evancz/virtual-dom 2.1.0

Do you approve of this plan? (y/n) y
Downloading elm-lang/core
Downloading evancz/elm-html
Downloading evancz/virtual-dom
Packages configured successfully!
$ cat elm-package.json
{
    "version": "1.0.0",
    "summary": "helpful summary of your project, less than 80 characters",
    "repository": "https://github.com/USER/PROJECT.git",
    "license": "BSD3",
    "source-directories": [
        "."
    ],
    "exposed-modules": [],
    "dependencies": {
        "elm-lang/core": "2.1.0 <= v < 3.0.0",
        "evancz/elm-html": "4.0.1 <= v < 5.0.0"
    },
    "elm-version": "0.15.1 <= v < 0.16.0"
}
```

Maybe a problem on the server side of things? I notice that this repo has been bumped to 2.0.2, but `elm-package` is trying to download 2.0.1 for me.

Thank you.

**Comment by jvoigtlaender** [comment]
It's not a server issue in so far as on my machine (Windows) I see this:
```shell
C:\Users\JanisVoigtlaender\Desktop\elm\test> elm-package install evancz/start-app
To install evancz/start-app I would like to add the following dependency to elm-package.json:

    "evancz/start-app": "2.0.1 <= v < 3.0.0"

May I add that to elm-package.json for you? (y/n) y

Some new packages are needed. Here is the upgrade plan.

  Install:
    elm-lang/core 2.1.0
    evancz/elm-effects 2.0.0
    evancz/elm-html 4.0.1
    evancz/start-app 2.0.1
    evancz/virtual-dom 2.1.0

Do you approve of this plan? (y/n) y
Downloading elm-lang/core
Downloading evancz/elm-effects
Downloading evancz/elm-html
Downloading evancz/start-app
Downloading evancz/virtual-dom
Packages configured successfully!
```
That the repo was bumped to `2.0.2` should be irrelevant, since `2.0.2` was not released to http://package.elm-lang.org/.

I had "hoped" that the issue at your end was some constraint conflict, but that this happens for you in an empty directory (like my experiment above is in an empty directory), pretty much rules that out.

**Comment by jvoigtlaender** [comment]
There is [this issue](https://github.com/elm-lang/elm-package/issues/142), which has an analogous error message. There, it comes from the `.json` being malformed, but I don't actually see any such problem in https://github.com/evancz/start-app/blob/2.0.1/elm-package.json.

**Comment by jvoigtlaender** [comment]
Another thought: It could be a parsing problem related to the `.json`, but not with the structure, but with the tokens/character encoding. Is there anything special about your system in that regard? You might be suffering from something like [this](https://github.com/elm-lang/elm-make/issues/33), and some of the solutions discussed over there might help you.

(I don't know why https://github.com/evancz/start-app/blob/2.0.1/elm-package.json would contain an unusually encoded character, whereas the `elm-package.json` files of the other packages you installed successfully do not. But it's possible.)

**Comment by jvoigtlaender** [comment]
Also, you could try to find out whether the issue is specific to version `2.0.1` of the package, by trying
```shell
elm-package install evancz/start-app 1.0.0
```

**Comment by ormsbee** [comment]
I'm running OS X 10.10.5 -- nothing particularly unusual in my setup. Trying start-app 1.0.0 does work. The JSON file looks like it's just ASCII. Maybe I'm getting a garbled version back when it tries to download it remotely? I'll poke at it more tonight. Thank you.

**Comment by ormsbee** [comment]
I tinkered with this more over the weekend and figured out what I was doing wrong. The first time I tried to install start-app via elm-package, I was at a hotel. It made an HTTP request for start-app's `elm-package.json` file, but the hotel wifi proxy caused that to load a sign-in page. But elm-package then took that response and saved it to `~/.elm/package/evancz/start-app/2.0.1/elm-package.json`. I didn't realize that file even existed locally like that.

So basically, it saw that I wanted to install 2.0.1, saw that it already had a cached version of that file in my local `~/.elm/`, and kept dying when trying to parse the hotel's login HTML as JSON. I'm closing this issue and I'll log a new bug with elm-package.

Thank you so much for your help on this. I really appreciate it.

**Comment by anthonylouisburns** [comment]
getting exact same error on osx

right now when trying to build the examples

I deleted teh repo and rechecked them out, still won't build, pretty sure it built at some point

**Comment by anthonylouisburns** [comment]
deleted dependencies from elm-package.json and reinstalled and it worked 

dependencies changed from     "dependencies": {
        "elm-lang/core": "3.0.0 <= v < 4.0.0",
        "evancz/elm-html": "4.0.1 <= v < 5.0.0",
        "evancz/start-app": "2.0.0 <= v < 3.0.0"
    },

to 

    "dependencies": {
        "elm-lang/core": "2.1.0 <= v < 3.0.0",
        "evancz/elm-html": "4.0.2 <= v < 5.0.0",
        "evancz/start-app": "2.0.2 <= v < 3.0.0"
    },

**Comment by lovasoa** [comment]
I've had the same issue for the same reason. I think elm-install shouldn't save the `elm-package.json` if it's invalid. And use trusted HTTPS connections. That would solve the problem.

**Comment by jvoigtlaender** [comment]
@lovasoa, note that you are commenting on a closed issue. Moreover, on a repository where the repository's owner has notifications turned off. Your comment is unlikely to have any effect that way. I suggest you open an issue (or two) at https://github.com/elm-lang/elm-package or https://github.com/elm-lang/elm-make, after checking there aren't already ones about writing invalid `elm-package.json` and/or about https vs http.

**Comment by lovasoa** [comment]
@jvoigtlaender Why is this closed? This is clearly a bug, isn't it?

https://github.com/elm-lang/elm-package/issues/194 : I think this is linked.

**Comment by jvoigtlaender** [comment]
It's closed here because it is not a bug of `evancz/start-app`, and never was.

**Comment by isaacsanders** [comment]
I am having this same issue, and I don't understand the solution. @jvoigtlaender Do you have a succinct answer to how I should fix this? I have tried deleting everything (./elm-stuff and ~/.elm) and that simply does not work.

**Comment by jvoigtlaender** [comment]
I am having this same issue, and I don’t understand the solution.
@jvoigtlaender <https://github.com/jvoigtlaender> Do you have a succinct
answer to how I should fix this? I have tried deleting everything
(./elm-stuff and ~/.elm) and that simply does not work.

What exactly does not work?
​

2016-04-20 17:16 GMT+02:00 Isaac Sanders <notifications@github.com>:
